File: VB.net/x9Rec.vb
Public Class x9Rec
  Private _recType As String
  Private _recData As String
  Private _recImage As String

  Sub New()

  End Sub

  Sub New(ByVal recType As String, ByVal recData As String, ByVal recImage As String)
    _recType = recType
    _recData = recData
    _recImage = recImage
  End Sub
  Public Property recType() As String
    Get
      Return _recType
    End Get
    Set(ByVal value As String)
      _recType = value
    End Set
  End Property

  Public Property recData() As String
    Get
      Return _recData
    End Get
    Set(ByVal value As String)
      _recData = value
    End Set
  End Property

  Public Property recImage() As String
    Get
      Return _recImage
    End Get
    Set(ByVal value As String)
      _recImage = value
    End Set
  End Property

End Class

Public Class x9Recs
  Inherits System.Collections.CollectionBase
  Public Function Add(ByVal newX9Rec As x9Rec) As Integer
    Return List.Add(newX9Rec)
  End Function
  Public Sub Remove(Optional ByVal index As Integer = -1)
    If index = -1 Then index = List.Count - 1
    If index >= 0 And index < List.Count Then
      List.RemoveAt(index)
    End If
  End Sub
  Default Public ReadOnly Property Item(ByVal index As Integer) As x9Rec
    Get
      Return CType(List.Item(index), x9Rec)
    End Get
  End Property
End Class

File: C#.net/MTS.ICLViewer/x9Rec.cs
//INSTANT C# NOTE: Formerly VB.NET project-level imports:
using System;
using System.Collections;
using System.Collections.Generic;
using System.Data;
using System.Drawing;
using System.Diagnostics;
using System.Windows.Forms;
using System.Linq;
using System.Xml.Linq;

namespace MTS.ICLViewer
{
    public class x9Rec
    {
        private string _recType;
        private string _recData;
        private string _recImage;

        public x9Rec()
        {

        }

        public x9Rec(string recType, string recData, string recImage)
        {
            _recType = recType;
            _recData = recData;
            _recImage = recImage;
        }
        public string recType
        {
            get
            {
                return _recType;
            }
            set
            {
                _recType = value;
            }
        }

        public string recData
        {
            get
            {
                return _recData;
            }
            set
            {
                _recData = value;
            }
        }

        public string recImage
        {
            get
            {
                return _recImage;
            }
            set
            {
                _recImage = value;
            }
        }

    }

    public class x9Recs : System.Collections.CollectionBase
    {
        public int Add(x9Rec newX9Rec)
        {
            return List.Add(newX9Rec);
        }

        public void Remove(int index = -1)
        {
            if (index == -1)
            {
                index = List.Count - 1;
            }
            if (index >= 0 && index < List.Count)
            {
                List.RemoveAt(index);
            }
        }
        public x9Rec this[int index]
        {
            get
            {
                return (x9Rec)(List[index]);
            }
        }
    }

} //end of root namespace